### PR TITLE
Add Freq() filter

### PR DESCRIPTION
**Pull request by bitfield**
Implements part of what's required for #7.

### CODE REVIEW

File: filters.go [hunk 1]
```diff
@@ -157,3 +159,38 @@ func (p *Pipe) First(lines int) *Pipe {
...
+// status is also set.
+func (p *Pipe) Freq() *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	var freq = map[string]int{}
+	p.EachLine(func(line string, out *strings.Builder) {
+		freq[line]++
+	})
+	type frequency struct {
```
Comment (breml) [inline filters.go]: We could check the len of the map and short cirquit the rest, if we have 0 elements in the map.

File: filters.go [hunk 2]
```diff
@@ -157,3 +159,38 @@ func (p *Pipe) First(lines int) *Pipe {
...
+		freqs = append(freqs, frequency{line, count})
+	}
+	sort.Slice(freqs, func(i, j int) bool {
+		if freqs[i].count == freqs[j].count {
+			return freqs[i].line < freqs[j].line
+		}
+		return freqs[i].count > freqs[j].count
+	})
+	var output strings.Builder
+	for _, item := range freqs {
```
Comment (breml) [inline filters.go]: As the slice contains the line with the highest fequency as the first element, we can use the length of the frequency in digits to prefix all lines accordingly with something along these lines (given, we check the len already on line 176):

```
digits := len(strconv.Itoa(freqs[0]))
for _, item := range freqs {
        output.WriteString(fmt.Sprintf(fmt.Sprintf("%% %dd %%s", digits), item.count, item.line))
```
```

File: testdata/freq.input.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apple
```
Comment (breml) [inline testdata/freq.input.txt]: The test case should at least produce double digit frequencies to test the correct prefixing of the single digit frequencies.

File: filters.go (hunk 1 above)
Comment (bitfield) [reply filters.go]: Yes, that's a good suggestion, but there's not actually a lot to short-circuit. The `for` loop will execute zero times, the `sort.Slice` will be a no-op, and then there'll be no output!

File: filters.go (hunk 2 above)
Comment (bitfield) [reply filters.go]: That's roughly what I ended up doing on the other branch:

```go
	fieldWidth := len(strconv.Itoa(maxCount))
	...
		output.WriteString(fmt.Sprintf("%*d %s", fieldWidth, item.count, item.line))
```

(I was quite pleased to discover the `%*` construct—exactly what I needed!)

File: testdata/freq.input.txt (hunk 3 above)
Comment (bitfield) [reply testdata/freq.input.txt]: Yes indeed. Already done on the other branch 👍

File: filters.go (hunk 2 above)
Comment (breml) [reply filters.go]: Great catch, i did not know about `%*`.

Comment (bitfield) [reply filters.go]: Neither did I—it's _almost_ undocumented, but it is there in the small print of `fmt`, if you use a large magnifying glass.

**Pull request merged**